### PR TITLE
Swerve autonomous updates

### PR DESCRIPTION
**Pull request by NogaRoze**
# Pull Request Details

## Description
Mirrors auton correctly for different alliances, essential for any auton routine
Adds a method to automatically align drivebase to a gyro heading, important for some teleop functions
Moves followTrajectory() to Swerve Subsystem
Adds the Field Constants class, to store information about the field
Adds new pathplanner paths

## Which does this affect?
AutonSubystem
SwerveSubsystem
pathplanner

## Testing Done
We tested each basic path (Forward, Backward, Forward180, Curve) for both alliances

## Checklist

- [x] My code follows the code style of this project.
- [x] My code has been tested on the Robot OR includes simulation code to prove functionality.

### CODE REVIEW

**Comment by azvanderpas** [comment]
> Please switch to **Preview** and select one of the following links:
> 
>     * [develop](?template=develop.md)
> 
>     * [main](?template=main.md)
> 
> 
> Once switched to the correct template, you can save the link as a bookmark. Keep in mind that switching templates will remove all already entered data within this issue.

You were meant to click the develop link to update the PR description with a template. Understandably confusing though...

This branch needs to be merged from develop or rebased onto develop. Currently there are a bunch of common commits between them.

**Comment by zbrotherton2158** [comment]
Updated PR to use the template and merged from develop

**Comment by caseyjbrotherton** [comment]
Interesting thing, mechanical advantages code is MIT license, which requires an accompanying license file.

https://mit-license.org/

So, need that before we merge

File: sysid/swerveConfig.json
Comment (azvanderpas) [inline sysid/swerveConfig.json:1]: Is this a WIP (Work in progress)? If so, we should probably move it out of this PR and add it when it is ready in a later one.

File: src/main/java/frc/robot/subsystems/AutonSubsytem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/AutonSubsytem.java:156]: For trajectories with multiple segments, does the resetOdometry method need to be called between them?

File: src/main/java/frc/robot/subsystems/SwerveSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/SwerveSubsystem.java:146]: Do we need a deadband on this?

**Review by azvanderpas — CHANGES_REQUESTED** [review]
Added paths should be tested on robot or in simulation before merge. Probably want to wait until we discuss which paths we want to have as a team and then update & test.

File: src/main/java/frc/robot/subsystems/AutonSubsytem.java
Comment (zbrotherton2158) [reply src/main/java/frc/robot/subsystems/AutonSubsytem.java:156]: No, reset odometry is called only once

File: src/main/java/frc/robot/subsystems/SwerveSubsystem.java
Comment (zbrotherton2158) [reply src/main/java/frc/robot/subsystems/SwerveSubsystem.java:146]: No, since r already uses the deadband set in OI

File: sysid/swerveConfig.json
Comment (zbrotherton2158) [reply sysid/swerveConfig.json:1]: It's a completed sysID configuration for the drivebase

File: vendordeps/PathplannerLib.json
Comment (azvanderpas) [inline vendordeps/PathplannerLib.json:4]: The version needs to be upgraded again. Seems to be why the CI workflow is breaking

**Review by azvanderpas — APPROVED** [review]

**Comment by azvanderpas** [comment]
We should hold off on merging this until after the meeting about Auton tomorrow

**Comment by zbrotherton2158** [comment]
> We should hold off on merging this until after the meeting about Auton tomorrow

Sorry, didn't see this message

**Pull request merged**
